hy/compiler.py

```diff
@@ -2312,6 +2316,19 @@ def compile_function_def(self, expression):
 
         return ret
 
+    @builds("return")
+    @checkargs(max=1)
+    def compile_return(self, expr):
+        value = None
+        ret = Result()
+        if len(expr) > 1:
+            value = self.compile(expr[1])
```

Comment:
Slightly bikeshedding, but here `value` is switching types which is kinda confusing. Usually in the compiler code, this would just be something like:

```python
compiled_expr = self.compile(expr[1])
ret += compiled_expr
value = compiled_expr.force_expr
```

or maybe another name still.